java-util/src/main/java/org/apache/druid/java/util/http/client/NettyHttpClient.java

```diff
@@ -242,22 +264,37 @@ public void messageReceived(ChannelHandlerContext ctx, MessageEvent e)
             }
           }
 
+          private void possiblySuspendReads(ClientResponse<?> response)
+          {
+            if (!response.isContinueReading()) {
+              synchronized (watermarkLock) {
+                suspendWatermark = Math.max(suspendWatermark, currentChunkNum);
+                if (suspendWatermark > resumeWatermark) {
+                  channel.setReadable(false);
```

Comment:
so, this is the magic way of telling netty to stop reading data off of socket without blocking any of the worker threads?

Reply:
Yes, it is. It makes netty stop reading from the socket and stop sending new data up the channel.